NuRadioReco/modules/io/NuRadioRecoio.py

```diff
@@ -221,8 +242,26 @@ def get_detector(self):
         event is given.
         """
         if self._current_file_id not in self.__detectors.keys():
+            detector_dict = self._detector_dicts[self._current_file_id]
```

Comment:
Can you add a few comments to the code saying what it being done here.

Reply:
Done